server/resp/src/main/java/org/infinispan/server/resp/commands/iteration/BaseIterationCommand.java

```diff
@@ -40,7 +41,7 @@ public final CompletionStage<RespRequestHandler> perform(Resp3Handler handler, C
       if (INITIAL_CURSOR.equals(cursor)) {
          CompletionStage<IterationInitializationContext> initialization = initializeIteration(handler, arguments);
          if (initialization != null) {
-            return initialization.thenCompose(iic -> initializeAndIterate(handler, ctx, manager, args, iic));
+            return handler.stageToReturn(initialization.thenCompose(iic -> initializeAndIterate(handler, ctx, manager, args, iic)), ctx, Function.identity());
```

Comment:
Instead of reusing Function.identity everywhere, lets change this instead to have a new method for `stageToReturn` that uses `exceptionallyAsync` with a cached lambda that doesn't need to allocate per call since there will be nothing required to capture.

Reply:
Updating with `handleAsync` because `exceptionallyAsync` is 12+ :/

Looking closer, some uses of `Function.identity()` weren't needed. Updating, too.